packages/@aws-cdk/aws-stepfunctions-tasks/lib/dynamodb/shared-types.ts

```diff
@@ -108,82 +110,131 @@ export class DynamoProjectionExpression {
+   * Sets a literal number. For example: 1234
+   * Numbers are sent across the network to DynamoDB as strings,
+   * to maximize compatibility across languages and libraries.
+   * However, DynamoDB treats them as number type attributes for mathematical operations.
+   */
+  public static wrapNumber(n: number) {
+    return new DynamoAttributeValue({ N: n.toString() });
+  }
 
   /**
    * Sets an attribute of type Number. For example:  "N": "123.45"
    * Numbers are sent across the network to DynamoDB as strings,
    * to maximize compatibility across languages and libraries.
    * However, DynamoDB treats them as number type attributes for mathematical operations.
    *
-   * @default - None
+   * Numbers may be expressed as literal strings or as JsonPath
    */
-  readonly n?: string;
+  public static wrapNumberFromString(n: string) {
+    return new DynamoAttributeValue({ N: n.toString() });
+  }
 
   /**
    * Sets an attribute of type Binary. For example:  "B": "dGhpcyB0ZXh0IGlzIGJhc2U2NC1lbmNvZGVk"
-   *
-   * @default - None
    */
-  readonly b?: string;
+  public static wrapBinary(b: string) {
+    return new DynamoAttributeValue({ B: b });
+  }
 
   /**
    * Sets an attribute of type String Set. For example:  "SS": ["Giraffe", "Hippo" ,"Zebra"]
-   *
-   * @default - None
    */
-  readonly ss?: string[];
+  public static wrapStringSet(ss: string[]) {
+    return new DynamoAttributeValue({ SS: ss });
+  }
+
+  /**
+   * Sets an attribute of type Number Set. For example:  "NS": ["42.2", "-19", "7.5", "3.14"]
+   * Numbers are sent across the network to DynamoDB as strings,
+   * to maximize compatibility across languages and libraries.
+   * However, DynamoDB treats them as number type attributes for mathematical operations.
+   */
+  public static wrapNumberSet(ns: number[]) {
+    return new DynamoAttributeValue({ NS: ns.toString() });
+  }
 
   /**
    * Sets an attribute of type Number Set. For example:  "NS": ["42.2", "-19", "7.5", "3.14"]
    * Numbers are sent across the network to DynamoDB as strings,
    * to maximize compatibility across languages and libraries.
    * However, DynamoDB treats them as number type attributes for mathematical operations.
    *
-   * @default - None
+   * Numbers may be expressed as literal strings or as JsonPath
    */
-  readonly ns?: string[];
+  public static wrapNumberSetFromString(ns: string[]) {
+    return new DynamoAttributeValue({ NS: ns });
+  }
 
   /**
    * Sets an attribute of type Binary Set. For example:  "BS": ["U3Vubnk=", "UmFpbnk=", "U25vd3k="]
-   *
-   * @default - None
    */
-  readonly bs?: string[];
+  public static wrapBinarySet(bs: string[]) {
+    return new DynamoAttributeValue({ BS: bs });
+  }
 
   /**
    * Sets an attribute of type Map. For example:  "M": {"Name": {"S": "Joe"}, "Age": {"N": "35"}}
-   *
-   * @default - None
    */
-  readonly m?: { [key: string]: DynamoAttributeValue };
+  public static wrapMap(m: { [key: string]: DynamoAttributeValue }) {
```

Comment:
What if *my whole map* is in a JSON path?

Reply:
good catch, adding a method to support jsonpath to be wrapped as a map